### PR TITLE
Support/v0.11

### PR DESCRIPTION
**Pull request by astein-te**
/review @jshields-te

### CODE REVIEW

File: gateways/core/resource/start.go [hunk 1]
```diff
@@ -126,6 +129,15 @@ func (executor *ResourceEventSourceExecutor) listenEvents(resourceCfg *resource,
 					executor.Log.WithField(common.LabelEventSource, eventSource.Name).WithError(err).Warnln("failed to apply the filter")
 					continue
 				}
+				normalizer, err := normalizerIgnoreDifferences(resourceCfg.Filter)
+				if err != nil {
+					executor.Log.WithField(common.LabelEventSource, eventSource.Name).WithError(err).Errorln("failed to generate ignoreDifferencesNormalizer")
+					continue
+				}
+				if err := hasNotChanged(normalizer, event); err != nil {
+					executor.Log.WithField(common.LabelEventSource, eventSource.Name).WithError(err).Warnln("dropping insignificant object update")
```
Comment (jshields-te) [inline gateways/core/resource/start.go]: Change `insignificant` to `ignored` to align with the configuration naming.

File: gateways/core/resource/config.go [hunk 2]
```diff
@@ -62,11 +62,12 @@ type resource struct {
...
-	Labels      map[string]string `json:"labels,omitempty"`
-	Annotations map[string]string `json:"annotations,omitempty"`
-	Fields      map[string]string `json:"fields,omitempty"`
-	CreatedBy   metav1.Time       `json:"createdBy,omitempty"`
+	Prefix            string            `json:"prefix,omitempty"`
+	Labels            map[string]string `json:"labels,omitempty"`
+	Annotations       map[string]string `json:"annotations,omitempty"`
+	Fields            map[string]string `json:"fields,omitempty"`
+	CreatedBy         metav1.Time       `json:"createdBy,omitempty"`
+	IgnoreDifferences []string          `json:"ignoreDifferences,omitempty"`
```
Comment (jshields-te) [inline gateways/core/resource/config.go]: I know we are trying to keep it simpler for the initial version, but is there a reason to not just make this `[]v1alpha1.ResourceIgnoreDifferences` in the first place? Then we could just use it directly and automatically support all types and not have the hardcoding.

I'd be OK to have that in a separate PR, though, so we can move forward.

File: gateways/core/resource/start.go [hunk 3]
```diff
@@ -243,16 +243,13 @@ func normalizerIgnoreDifferences(filter *ResourceFilter) (diff.Normalizer, error
 }
 
 // helper method to see if resource is still "changed" after normalization for ignoreDifferences objects
-func hasNotChanged(normalizer diff.Normalizer, event *InformerEvent) error {
+func hasNotChanged(normalizer diff.Normalizer, event *InformerEvent) bool {
 	if normalizer == nil || event.OldObj == nil {
-		return nil
+		return true
```
Comment (jshields-te) [inline gateways/core/resource/start.go]: I think this should return `false`.

If the `normalizer` is nil that indicates that the configuration for ignoring differences was not provided. In that case I think we should treat the event as a change event and therefore return `false`, it *has* changed.

I think there is a little discussion here if the `OldObj` is nil. I *think* this also should return `false` to indicate it *has* changed considering it is presumably new (no old object). But I'm not 100% sure if there are not other cases where the old object would be nil.

File: gateways/core/resource/start.go [hunk 4]
```diff
@@ -126,6 +129,15 @@ func (executor *ResourceEventSourceExecutor) listenEvents(resourceCfg *resource,
 					executor.Log.WithField(common.LabelEventSource, eventSource.Name).WithError(err).Warnln("failed to apply the filter")
 					continue
 				}
+				normalizer, err := normalizerIgnoreDifferences(resourceCfg.Filter)
+				if err != nil {
+					executor.Log.WithField(common.LabelEventSource, eventSource.Name).WithError(err).Errorln("failed to generate ignoreDifferencesNormalizer")
+					continue
+				}
```
Comment (jshields-te) [inline gateways/core/resource/start.go]: I think this code should be moved outside of the `go func` block. The resource config filter never changes so there shouldn't be a need (I think) to keep creating a new normalizer for each event.

I think the error handling would also be changed to:

```go
		if err != nil {
			errorCh <- err
			return
		}
```

File: gateways/core/resource/start.go (hunk 3 above)
Comment (astein-te) [reply gateways/core/resource/start.go]: I agree with you on both points - issuing change.

File: gateways/core/resource/start.go (hunk 4 above)
Comment (astein-te) [reply gateways/core/resource/start.go]: done.
